scripts/benchmark_multithreaded.py

```diff
@@ -0,0 +1,205 @@
+    if max_core_count < 8:
+        step = -1
+    elif max_core_count < 16:
+        step = -int(max_core_count / 4)
+    else:
+        step = -int(max_core_count / 8)
+    core_counts = list(range(max_core_count, 0, step))
+    if core_counts[-1] != 1:
+        core_counts.append(1)
+    core_counts.append(0)
+    return core_counts
+
+def get_formatted_queries(args):
+    queries = args.queries if args.queries else DEFAULT_TPCH_QUERIES
+    query_ids = ','.join([str(query) for query in queries])
+    return ['--queries', query_ids]
+
+def is_square(n):
+    return n ** 0.5 == int(n ** 0.5)
+
+def get_subplot_row_and_column_count(num_plots):
+    while not is_square(num_plots):
+        num_plots += 1
+    return num_plots ** 0.5
+
+def run_benchmarks(args, hyrise_args, core_counts, result_dir):
+    benchmark_run = 0
+    for core_count in core_counts:
+        benchmark_run += 1
+        verbose_print(args.verbose, 'Starting benchmark run ' + str(benchmark_run) + ' of ' + str(len(core_counts)) + ' ...')
+
+        if core_count == 0:
+            use_scheduler = 'false'
+            number_of_runs = args.runs_per_core if args.runs_per_core else args.runs
+        else:
+            use_scheduler = 'true'
+            number_of_runs = args.runs_per_core * core_count if args.runs_per_core else args.runs
+
+        file_name = str(core_count) + '-cores.json'
+        result_file = os.path.join(result_dir, file_name)
+        execution_command = [
+            args.executable,
+            '--output', result_file,
+            '--runs', str(number_of_runs),
+            '--scheduler', use_scheduler,
+            '--cores', str(core_count),
+            '--clients', str(args.clients),
+        ]
+        if args.verbose:
+            execution_command.append('--verbose')
+        if hyrise_args:
+            execution_command += hyrise_args
+        execution_command += get_formatted_queries(args)
+        verbose_print(args.verbose, 'Executing command: ' + subprocess.list2cmdline(execution_command) + '\n')
+        subprocess.run(execution_command)
+
+def benchmark(args, hyrise_args):
+    result_name = args.result_name if args.result_name else datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+    result_dir = os.path.join(args.result_dir, result_name)
+        
+    if not os.path.exists(result_dir):
+        os.makedirs(result_dir)
+
+    core_counts = get_core_counts(args)
+    verbose_print(args.verbose, 'Executing benchmarks with the following numbers of cores: ' + str(core_counts))
+
+    # Make sure our machine has the number of cores we are trying to benchmark with
+    for core_count in core_counts:
+        assert core_count <= MAX_CORE_COUNT, 'Can\'t use more cores than we have available'
+
+    run_benchmarks(args, hyrise_args, core_counts, result_dir)
+
+    verbose_print(args.verbose, 'Benchmarks complete!')
+    verbose_print(args.verbose, 'Saved JSON results to: ' + result_dir)
+
+    return result_dir
+
+def plot(args, result_dir):
```

Comment:
Can we get the NUMA boundaries in?

Reply:
Yep, I added the number of cores per NUMA node to the result JSON output, and with that the python script can print the boundaries (see updated image below).